### ISSUE TITLE
Redwoods spawn too close to edges

### ISSUE THREAD

**Issue by D3nnis3n** [StrangeLoopGames/EcoIssues#11142]
![redwood](https://user-images.githubusercontent.com/25908592/52409885-129b1780-2ad7-11e9-8e95-86619d804120.png)

**Comment by Deantwo** [comment]
Unless it literally right on the edge, it kinda looks good with the roots going out everywhere.

**Comment by keegan-o** [comment]
I don't think we want to further alter the art to try to account for the million and one situations it could encounter, is this something that could be solved in code? @craig-jong

**Comment by D3nnis3n** [comment]
I find it ugly. And its part of qa test. So thats why i Reported it.

Am Di., 12. Feb. 2019, 22:48 hat keegan-o <notifications@github.com>
geschrieben:

**Comment by elfl0rd** [comment]
I believe that small amount of roots can be visible, but not to such an extent as in the screenshot. I support - the screenshot looks ugly.

**Comment by Deantwo** [comment]
Might look a little silly, because it looks like the tree could fall at any time, but I would not call it ugly.

**Comment by beastyn** [comment]
[0.8.1.3] Not only on the edge. but in the cliffs
![bandicam 2019-04-18 15-56-18-793](https://user-images.githubusercontent.com/27898520/56356610-c9f88c80-61f2-11e9-97be-087892360679.jpg)

**Comment by Deantwo** [comment]
I would say that both cases, on the edge or in a cliff, could happen in real life. So what is the issue?
Making special snowflake code to prevent these things seem like a waste of time.

But solution would be more along the lines of making roots look differently when hanging over an edge or when a tree in up against a cliff side. But that is likely more of an art improvement than a code issue.

**Comment by D3nnis3n** [comment]
![redwood](https://user-images.githubusercontent.com/25908592/56398426-b8da6a80-6248-11e9-842e-99d215b8bcd9.png)
Please don't do that!

**Comment by MazonDel** [comment]
The current solution to this issue is in the branch titled "Bug-11142-Implementation". It was branched off of the "Develop" branch on 7/28/2020.

This adds a "Width" parameter to trees and alters the IsGoodPlacement() method to check the surrounding Blocks for validity conditions (ex: If there is empty space, that is not allowed. If there's a plant/tree there, that's not allowed.). In addition, this adds the IsGoodPlacement() check to the WorldLayerManager.PostWorldGenPlants() method, which did not previously check if the position was valid for the plant in question.

Note: To Implement these features you will need to add a Width parameter to the trees in the EcoTechTree CSV, or (temporarily) manually add the width parameter to the relevant tree classes.

**Comment by Quadbro** [comment]
@MazonDel Can you ask @WeaselDog or @keegan-o to add that parameter to tech tree and implement that fix?
